src/container/acrManager.ts

```diff
@@ -103,7 +103,12 @@ export class AcrManager {
     }
 
     private async selectAcrRepo(registryUrl: string, session: AzureSession): Promise<vscode.QuickPickItem> {
-        const acrRepoItem: vscode.QuickPickItem = await vscode.window.showQuickPick(this.loadAcrRepoItems(registryUrl, session), { placeHolder: "Select Repository", ignoreFocusOut: true });
+        const repoItems = await this.loadAcrRepoItems(registryUrl, session);
+        if (!repoItems) {
+            vscode.window.showWarningMessage("There is no repository in the registry.");
+            return;
+        }
+        const acrRepoItem: vscode.QuickPickItem = await vscode.window.showQuickPick(repoItems, { placeHolder: "Select Repository", ignoreFocusOut: true });
```

Comment:
FYI, by updating the parameter passed to `showQuickPick` from a `Promise` to an `array`, VS Code will not show loading animations.

Reply:
Good catch. @SLdragon, how about keep "promise", and return empty list of quickpickitem when there is no repository

Reply:
Another option is to simply throw a 404 error and https://github.com/Microsoft/vscode-azure-iot-edge/pull/412/files#diff-36614ddf00470008048ba066990c9c6fR145 will handle it.